sgkit/stats/popgen.py

```diff
@@ -121,9 +121,7 @@ def diversity(
 
 # c = cohorts, k = alleles
 @guvectorize(  # type: ignore
```

Comment:
Is there any good reason not to have ``cache=True`` for our use-case? Perhaps it's worth us having our own local ``guvectorize`` decorator that does this by default?

This is the sort of thing that's hard to test for and easy to forget, so maybe it's worth a little indirection.

Reply:
Maybe? It's not hard to add and now we have the convention started it is likely to be propagated in our code.

Reply:
Opened #371